### PR TITLE
SAK-45481 Preserve rubric associations while attaching files

### PR DESCRIPTION
**Pull request by adrianfish**
https://jira.sakaiproject.org/browse/SAK-45481

### CODE REVIEW

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 1]
```diff
@@ -6992,6 +6995,21 @@ private void setNewAssignmentParameters(RunData data, boolean validify) {
             }
         }
 
+        String rubricId = params.getString(RubricsConstants.RBCS_LIST);
+        if (StringUtils.isNotBlank(rubricId)) {
+            Map<String, Object> rubricAssociationMap = new HashMap<>();
+            rubricAssociationMap.put("rubricId", params.getString(RubricsConstants.RBCS_LIST));
```
Comment (bjones86) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java:7001]: I think you already pulled this value out of `params` on line 6998.

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-association.js [hunk 2]
```diff
@@ -89,7 +107,7 @@ class SakaiRubricAssociation extends RubricsElement {
           <div class="rubrics-selections">
             <select @change="${this.rubricSelected}" name="rbcs-rubricslist" aria-label="${tr("rubric_selector_label")}" class="form-control" ?disabled=${!this.isAssociated || this.readOnly}>
             ${this.rubrics.map(r => html`
-              <option value="${r.id}" ?selected=${r.id === this.selectedRubric}>${r.title}</option>
+              <option value="${r.id}" ?selected=${r.id == this.selectedRubric}>${r.title}</option>
```
Comment (bjones86) [inline webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-association.js:110]: Is this change intentional? I always try to avoid coercion by using `===` rather than `==`

**Review by bjones86 — APPROVED** [review]
LGTM, just a couple minor things

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-association.js (hunk 2 above)
Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-association.js:110]: It was intentional as that selectedRubric might have been coming from an attribute. I think I changed that though. Good spot.
